### PR TITLE
Print default target device

### PR DESCRIPTION
**Pull request by YoungeunKim**
:Release Notes:
Avoid multiple session connections in CLI

:Detailed Notes:
- Fix multiple ssh connection
- Change ares-shell's internal option name to device
- Remove unused code
- Update TC

:Testing Performed:
1. Passed unit test ose & auto target
2. Passed eslint
3. Verified with CLI commands 
a. `$killall node`
b. Check ssh connection using `$ss |grep ssh` . It shows no list.
c. Install com.jasmine.web.app in CLI TC
d. `$ares-inspect.js -s com.jasmine.web.app.service` 
e. `$ss |grep ssh` shows only one ssh connection.
f. Press Ctrl+C to exit ares-shell.
g. `$ss |grep ssh` shows no list.

:Issues Addressed:
[PLAT-137688] Avoid multiple ssh connections in CLI

### CODE REVIEW

**Comment by YoungeunKim** [comment]
Code 👍 
Verified 👍 

Need to have a discussion about the added information sentence("Set target device to oseTarget) with @ssuminahn .

File: lib/base/novacom.js [hunk 1]
```diff
@@ -334,48 +334,6 @@ const fs = require('fs'),
             }
         },
 
-        getSshPrvKey: function(target, next) {
```
Comment (ssuminahn) [inline lib/base/novacom.js:337]: Did you delete this code because this code is not used no longer?

File: lib/device.js [hunk 2]
```diff
@@ -55,7 +55,11 @@ const util = require('util'),
 
             function _makeSession(next) {
                 options.nReplies = 1; // -n 1
-                options.session = new novacom.Session(options.device, next);
+                if (!options.session) {
```
Comment (ssuminahn) [inline lib/device.js:58]: I think the old code is better.
Rather than checking the session and passing it over
I think it's best to create a new one, as it's certainly best to create a new one and there may be cases where you are using an old session.

File: lib/install.js [hunk 3]
```diff
@@ -253,11 +252,7 @@ const fs = require('fs'),
                     }
                 },
                 function(next) {
-                    options.session.end();
-                    options.session = null;
-                    next(null, {
-                        msg: "Success"
-                    });
+                    next(null, {msg: "Success"});
```
Comment (ssuminahn) [inline lib/install.js:255]: msg : "Success"

File: lib/install.js [hunk 4]
```diff
@@ -273,13 +268,16 @@ const fs = require('fs'),
...
+                    } else {
+                        next(null, options.session);
+                    }
                 },
                 function(session, next) {
-                    options.session = session;
                     if (options.opkg) {
                         // FIXME: Need more consideration whether this condition is necessary or not.
-                        if (options.session.getDevice().username !== 'root') {
+                        if (session.getDevice().username !== 'root') {
```
Comment (ssuminahn) [inline lib/install.js:280]: #272, options.session = new novacom.Session(options.device, next); 
already options.session is assigned.
I prefer 
if (options.session.getDevice().username !== 'root') {

File: spec/jsSpecs/ares-device.spec.js [hunk 5]
```diff
@@ -143,8 +143,6 @@ describe(aresCmd + ' --capture-screen(-c)', function() {
             if (stderr && stderr.length > 0) {
                 common.detectNodeMessage(stderr);
             }
-            expect(stdout).not.toContain(options.device);
```
Comment (ssuminahn) [inline spec/jsSpecs/ares-device.spec.js:146]: why do you remove this codes?

File: spec/jsSpecs/ares-inspect.spec.js [hunk 6]
```diff
@@ -98,7 +97,7 @@ describe(aresCmd, function() {
 
         setTimeout(() => {
             child.kill();
-            expect(result).not.toBeNull();
+            expect(result).toContain('Application Debugging - http://localhost');
```
Comment (ssuminahn) [inline spec/jsSpecs/ares-inspect.spec.js:100]: Was there a bug in this part? Why did you change it?
I think it's correct to check this when stdout is received.

if received error, this line is not meaningful.

File: spec/jsSpecs/ares-inspect.spec.js [hunk 7]
```diff
@@ -138,7 +136,7 @@ describe(aresCmd +' --open(-o)', function() {
...
```
Comment (ssuminahn) [inline spec/jsSpecs/ares-inspect.spec.js:139]: above same opnion.

File: spec/jsSpecs/ares-inspect.spec.js [hunk 8]
```diff
@@ -152,66 +150,61 @@ describe(aresCmd +' --open(-o)', function() {
...
-            result = data;
+            result += data;
             expect(data).toBeNull();
         });
 
         setTimeout(() => {
             child.kill();
-            expect(result).not.toBeNull();
+            expect(result).not.toContain("null");
+            expect(result).toContain("localhost");
```
Comment (ssuminahn) [inline spec/jsSpecs/ares-inspect.spec.js:180]: above same opinion.

File: spec/jsSpecs/ares-inspect.spec.js [hunk 9]
```diff
@@ -152,66 +150,61 @@ describe(aresCmd +' --open(-o)', function() {
...
             if (data && data.length > 0) {
                 common.detectNodeMessage(data);
             }
-            result = data;
             expect(data).toBeNull();
         });
 
         setTimeout(() => {
             child.kill();
+            expect(result).toContain("To debug your service, set \"localhost");
```
Comment (ssuminahn) [inline spec/jsSpecs/ares-inspect.spec.js:206]: above same opinion.

File: spec/jsSpecs/ares-shell.spec.js [hunk 10]
```diff
@@ -123,7 +123,7 @@ describe(aresCmd + ' --run in session', function() {
                     expect(stderr).toContain("ares-shell ERR! [Tips]: This device does not support multiple sessions");
                 }
             } else {
-                expect(stdout.trim()).toBe("hello webOS", stderr);
+                expect(stdout.trim()).toContain("hello webOS", stderr);
```
Comment (ssuminahn) [inline spec/jsSpecs/ares-shell.spec.js:126]: why do you change?
tobe is more strong

**Review by ssuminahn — COMMENTED** [review]
code :-1:

File: spec/jsSpecs/ares-install.spec.js [hunk 11]
```diff
@@ -80,6 +80,7 @@ describe(aresCmd + ' --list(-l)', function() {
             if (stderr && stderr.length > 0) {
                 common.detectNodeMessage(stderr);
             }
+            expect(stdout).toContain(`Set target device to ${options.device}`);
```
Comment (ssuminahn) [inline spec/jsSpecs/ares-install.spec.js:83]: only ares-install is affected?
or, do you update tcs after fixed sentence?

File: lib/base/novacom.js (hunk 1 above)
Comment (YoungeunKim) [reply lib/base/novacom.js:337]: Sure.

File: lib/device.js (hunk 2 above)
Comment (YoungeunKim) [reply lib/device.js:58]: I think that one ssh connection for one target connection is enough.
In the meantime, there were cases in which up to 3 sessions were connected.

This code is a code that prevents connecting multiple sessions.

File: lib/install.js (hunk 3 above)
Comment (YoungeunKim) [reply lib/install.js:255]: I'll fix it.

File: lib/install.js (hunk 4 above)
Comment (YoungeunKim) [reply lib/install.js:280]: No problem to use "options.session.getDevice". 
However, function(session) 's session value becomes "unused value".

File: spec/jsSpecs/ares-device.spec.js (hunk 5 above)
Comment (YoungeunKim) [reply spec/jsSpecs/ares-device.spec.js:146]: I added "Set target device to oseTarget" for this commit.
"expect(stdout).not.toContain(options.device)" makes always failure.

File: spec/jsSpecs/ares-inspect.spec.js (hunk 6 above)
Comment (YoungeunKim) [reply spec/jsSpecs/ares-inspect.spec.js:100]: When the result is null, "toContain" would be failed. 

"toContain" can check the "result" is not null. right?

File: spec/jsSpecs/ares-shell.spec.js (hunk 10 above)
Comment (YoungeunKim) [reply spec/jsSpecs/ares-shell.spec.js:126]: I added "Set target device to [devivcename]" for this commit.
 expect(stdout.trim()).toBe("hello webOS", stderr) makes always error.

File: spec/jsSpecs/ares-install.spec.js (hunk 11 above)
Comment (YoungeunKim) [reply spec/jsSpecs/ares-install.spec.js:83]: All effects to ares-commands which is to connect target. 

I put it representatively here. If you want to put it in all ares-commands that have an effect, I'll apply it to one TC on each command. I think the rest will be in common.

File: lib/base/novacom.js [hunk 12]
```diff
@@ -323,6 +322,7 @@ const fs = require('fs'),
             if (devices.length < 1) {
                 setImmediate(next, errHndl.getErrMsg("UNMATCHED_DEVICE", key, value));
             } else {
+                console.log("Set target device to " +  value);
```
Comment (YoungeunKim) [inline lib/base/novacom.js:325]: move position "Connecting target"

File: lib/device.js (hunk 2 above)
Comment (YoungeunKim) [reply lib/device.js:58]: Keep it.

File: lib/install.js (hunk 4 above)
Comment (YoungeunKim) [reply lib/install.js:280]: Change "options.session"

Comment (YoungeunKim) [reply lib/install.js:280]: Changed.

File: spec/jsSpecs/ares-inspect.spec.js (hunk 7 above)
Comment (YoungeunKim) [reply spec/jsSpecs/ares-inspect.spec.js:139]: the same command as above

File: spec/jsSpecs/ares-inspect.spec.js (hunk 8 above)
Comment (YoungeunKim) [reply spec/jsSpecs/ares-inspect.spec.js:180]: the same command as above

File: spec/jsSpecs/ares-inspect.spec.js (hunk 9 above)
Comment (YoungeunKim) [reply spec/jsSpecs/ares-inspect.spec.js:206]: the same command as above

**Comment by YoungeunKim** [comment]
Need to change not to show "ares-shell".

File: lib/install.js [hunk 13]
```diff
@@ -72,14 +72,13 @@ const fs = require('fs'),
             async.waterfall([
                 function(next) {
                     options.nReplies = 0; // -i
-                    // eslint-disable-next-line no-new
-                    new novacom.Session(options.device, next);
+                    if(!options.session){
```
Comment (YoungeunKim) [inline lib/install.js:75]: if (!options.session)

File: spec/jsSpecs/ares-install.spec.js (hunk 11 above)
Comment (YoungeunKim) [reply spec/jsSpecs/ares-install.spec.js:83]: I removed this logic.

File: lib/base/novacom.js (hunk 12 above)
Comment (YoungeunKim) [reply lib/base/novacom.js:325]: Removed this print. I'll add the device target information in CLI status message format work.
